### ISSUE TITLE
how to debug mach-nix

### ISSUE THREAD

**Issue by InLaw** [DavHau/mach-nix#93]
I would like to install pkgs but in combination there are different issues 
- how can I debug those with mach-nix 

```
let
   mach-nix = import (builtins.fetchGit {
     url = "https://github.com/DavHau/mach-nix/";
     ref = "refs/tags/2.2.2";
     rev = "f0b3b3c8e5153f61db7dcfbe2120c8d3e0e41371";
   }); 

  overlays = []; # some very useful overlays
  pkgs = import mach-nix.nixpkgs.path { config = { allowUnfree = true; }; inherit overlays; };
in mach-nix.mkPythonShell rec {
  python = pkgs.python37;
  requirements =  ''  
    aiohttp 
    arcgis 
    area 
    asyncio 
    autoflake 
    #avro 
    black
    bokeh 
    bottle 
    chart-studio 
    clang 
    conda 
    cython
    dash 
    dash-leaflet 
    dask 
    #dbt 
    #dephell 
    descartes 
    django 
    earthengine-api 
    eli5 
    fastapi 
    Fiona 
    flake8 
    flask 
    folium 
    fuzzywuzzy
    GeoAlchemy2 
    geofeather 
    geojsonio 
    geopandas 
    geopy 
    geotable 
    gprof2dot 
    graphviz 
    h2o 
    holoviews 
    hunter 
    hvplot 
    imbalanced-learn 
    ipdb 
    ipyparallel 
    ipython 
    ipywidgets 
    pythreejs
    jq 
    json2json 
    json2yaml 
    jupyter_contrib_nbextensions 
    #jupyterhub 
    jupyterlab>=2.2.0
    jupyter_nbextensions_configurator 
    lidar 
    lime 
    line-profiler 
    Markdown 
    missingno 
    momepy 
    more-itertools 
    mypy 
    networkx 
    #Nikola # issues 
    objgraph 
    ogr2wkt 
    openpyxl 
    orange3 
    osmapi 
    osm-diff-tool 
    osmnet 
    overpass 
    #overpy2 
    pandana 
    pandas 
    streamz
    #panel 
    parquet 
    pip 
    pip-tools 
    pipx 
    #plop 
    plotly 
    #poetry2conda 
    pprint 
    pprofile 
    #pyarrow 
    #pycsw 
    pydal 
    pygeos
    pygis 
    pygments 
    pyinstrument 
    pymc3 
    Pympler 
    pyperf 
    pyproj 
    pyshp 
    py-spy 
    pytest 
    #pywps 
    quinn 
    rasterio 
    rasterstats 
    recurse 
    reportlab 
    #rio-cogeo 
    #rsgis 
    scikit-learn 
    seaborn 
    shap 
    Shapely 
    simplejson 
    spatial 
    stackimpact 
    tensorflow 
    tifffile 
    torch 
    tqdm 
    tuna 
    urbansim 
    verde 
    VisiData 
    # voila 
    vprof 
    #whitebox 
    widgetsnbextension 
    xarray 
    #xarray-spatial 
    xdot 
    xgboost 
    XlsxWriter 
    fuzzywuzzy
    spacy
    nltk 
    gensim
    #word2vec 
    quepy 
    textblob 
    Pattern 
    PyNLPl 
    corenlp
    #polyglot
    keplergl
    mapboxgl
    #osmnx   
    urbanpy 
    earthpy
    pysal
    gdal 
    xarray-leaflet
    #pypostal
    #vaex-jupyter
    pickle5
    pylint
    jupyterlab_pygments  
    jupyterlab_code_formatter  
    #libxml2python3
    nbresuse
    jupyterlab-git
    pyfxa
    pycrypto
    scikit-image
    ipykernel
    nbdime
    ipyleaflet
    jupyterlab_gitlab
    jupyterlab_latex
    isort
    jupyterlab_code_formatter
    jupyter_bokeh
    ipysheet
    xeus-python
    ptvsd
    pydevd-pycharm
    pylantern
    jupyterlab_commands
    ipympl
    jupyter-lsp  
    jupyter-kite
    ipymonaco
    jupyter-kite
    '';
}
```

**Comment by DavHau** [comment]
You environments builds with the newest version (2.3.0) if I remove `pprint` which has been deleted from pypi and add the following provider config:
```nix
  providers = {
    numpy = "nixpkgs";
    orange3 = "sdist";
  };
```
